cmd/sysl/cmd_datamodel.go

```diff
@@ -70,6 +108,35 @@ func generateDataModel(pclass ClassLabeler, outmap map[string]string, mod *sysl.
 	}
 }
 
+func generateDataModels(datagenParams *CmdContextParamDatagen,
+	model *sysl.Module, logger *logrus.Logger) (map[string]string, error) {
+	if !projectManner(model) {
+		// The sysl file is not project manner
+		return generateDataModelsWithPureModule(datagenParams, model, logger)
+	}
+	// Sysl file is project manner
+	return generateDataModelsWithProjectMannerModule(datagenParams, model, logger)
+}
+
+/**
+ * Check if sysl is project manner or not
+ */
+func projectManner(model *sysl.Module) bool {
+	var onlyApp *sysl.Application
+	apps := model.GetApps()
+	if len(apps) == 1 {
+		for _, app := range apps {
+			onlyApp = app
+		}
+	}
+
+	if onlyApp != nil && onlyApp.GetEndpoints() == nil {
+		return false
+	}
```

Comment:
Sorry, I should have dug into this myself earlier. The above logic isn't safe because data models can coexist with endpoints under the same application. We need to explicitly call out whether we're using the project structure or work directly against a model. On the face it, this suggests that your original implementation was correct.

Reply:
Ok, I will add an option to make sure it is working against a model directly.

Reply:
Add new option -d to explicitly identify it will process sysl file without project manner.